apps/language_server/test/providers/folding_range_test.exs

```diff
@@ -0,0 +1,408 @@
+defmodule ElixirLS.LanguageServer.Providers.FoldingRangeTest do
```

Comment:
Some general comments about your test structure:
1. The first four describes ("indentation", "token pairs", "heredocs", "comment blocks") are clearly testing sub-providers (ie not the `FoldingRange` module itself). I'd move their respective tests in separate files.
2. I'd get rid of your setup functions. Imo the tested behaviour should be clearly laid out in the test itself.

Reply:
> The first four describes ("indentation", "token pairs", "heredocs", "comment blocks") are clearly testing sub-providers (ie not the FoldingRange module itself). I'd move their respective tests in separate files.

I agree with that! I actually considered doing it myself. I just, for whatever reason, wanted to keep the # of test files I added small. IDK why, really.

However, I'm going to keep it as a single test file for now so I can keep my `defp`s in one place (see below). I might come back and do this later, though.

> I'd get rid of your setup functions. Imo the tested behaviour should be clearly laid out in the test itself.

Stylistically I'm on the opposite side of the spectrum. I like setting up little DSLs in my test suites because I think it reads better. If I have 10 tests that all test the roughly same thing, but I need the same 5 setup functions and asserts in each test, it's just noise to my brain.

Reply:
Yeah I get your point about having a small DSL to set up the tests. Personally I'd still rather call my one-liner setup function inside the test itself (unless it's really more of a global, unrelated setup), but it comes down to personal preference.